ui/src/main.mjs

```diff
@@ -175,4 +169,9 @@ fetch('_setup')
     })
     .catch((err) => {
         console.log('auth error:', err)
+        console.log('redirecting to:', window.location.origin + '/dashboard')
+
+        // window.location.replace(window.location.origin + '/dashboard') //Original, seems to have no issues with Edge and Chrome on Windows, doesn't work on Android (Not tested on Linux browser)
+        // window.location.href = window.location.origin + window.location.pathname + window.location.search + (window.location.search ? '&' : '?') + 'reloadTime=' + Date.now().toString() + window.location.hash; // Also works on Edge + Chrome on windows, doesn't work on android
+        window.location.replace(window.location.origin + '/dashboard' + '?' + 'reloadTime=' + Date.now().toString() + Math.random()) // Seems to work on Edge and Chrome on Windows Chromium and Firefox on Linux, and also on Chrome Android (an also as PWA App)
```

Comment:
@cstns  The error that occurs for me with `no-cors` at the following line is this:
```javascript
const r = new URL(e.url).pathname.replace("/_setup", "")
``` 

```
TypeError: Failed to construct 'URL': Invalid URL at https://nodered.mydomain.com/dashboard/assets/index-Cl3n5Iiw.js:239:407585
message: "Failed to construct 'URL': Invalid URL"
``` 

### Without `no-cors` it occurs at:

```javascript
fetch('_setup')
``` 

```
TypeError: Failed to fetch at https://nodered2.cgiothub.ddnsfree.com:1020/dashboard/assets/index-DvjY5ADf.js:239:425401
message: "Failed to fetch"
```

Would you suggest something like this in error handler: 


```javascript
if (error instanceof TypeError) {
    if (error.message === "Failed to construct 'URL': Invalid URL") {
      // handle specific "Invalid URL" error
      console.error("Invalid URL error occurred:", error);
    } else if (error.message === "Failed to fetch") {
      // handle specific "Failed to fetch" error
      console.error("Failed to fetch error occurred:", error);
    }
  } else {
    // handle general errors here
    console.error("An error occurred:", error);
  }
```

Reply:
Before making any changes to the catch redirect, please try another thing. 

Based on the things you mentioned, I have a feeling the whole setup might be failing because we're receiving an invalid response.url from the _setup call which causes `ui/src/main.mjs:71` to fail.

Would you be able to log the response and check if there's anything out of the ordinary with it?

I assume that due to what might be a failing cors request we might be receiving something we're not expecting on the response and passing it to the URL constructor.

Reply:
@cstns 
### With `no-cors`:

`Response {type: 'opaque', url: '', redirected: false, status: 0, ok: false, …}`

So, url is an empty string causing the error it seems.

### Without `no-cors`:

I do get this error again:

`
Access to fetch at 'https://auth.mydomain.com' (redirected from 'https://nodered.mydomain.com/dashboard/_setup') from origin 'https://nodered.mydomain.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
`
but the error handler still catches the error this time resulting in the desired behavior of redirecting to dashboard.

Error being caught here is: 
`TypeError: Failed to fetch at https://nodered.mydomain.com/dashboard/assets/index-DvjY5ADf.js:239:425401
message: "Failed to fetch"`

Regardless of whether the `no-cors` setting is enabled or not, it seems to be functioning correctly. My thinking would be to implement conditional logic to accommodate both cases, ensuring redirection is handled appropriately in either scenario.

Reply:
I think it's functioning correctly because of a fluke!

So the flow as I understand it is as follows:
Vue app boots up
 -> makes fetch call to _setup
 -> auth session lost so redirected to auth provider on cross domain
 
 from here we diverge in two sub flows:
 
 1. with cors enabled
  -> the cors error is caught by catch
  -> error is logged, thus the cors error you provided
  -> page location is replaced with cache bust
  -> taken to auth login, all is well
  
  2. without cors i.e. `no-cors`
   -> the cors error returned as a valid response, but not the response we need to complete the setup steps
   -> at line 71 we're passing null to the URL constructor which fails
   -> the error is caught by catch
   -> error logged and page location is replaced with cache bust
   -> taken to auth login
   
The fetch spec seems to confirm this:
![image](https://github.com/FlowFuse/node-red-dashboard/assets/20094749/5f768c12-33dc-4c6b-96d4-760872ca60a2)
From what understand from the docs, an [opaque filtered response](https://fetch.spec.whatwg.org/#concept-filtered-response-opaque) for sub-flow 2 means a successful response 
 
I'd treat the cors [network error](https://fetch.spec.whatwg.org/#concept-network-error) out of the two and redirect with cache bust _only_ in that scenario. But I don't have enough experience with the dashboard just yet and the redirect would serve multiple ailments